core/dbt/contracts/publication.py

```diff
@@ -0,0 +1,110 @@
+from typing import Optional, List, Dict, Any
+from datetime import datetime
+from dbt.dataclass_schema import dbtClassMixin
+
+from dataclasses import dataclass, field
+
+from dbt.contracts.util import BaseArtifactMetadata, ArtifactMixin, schema_version
+from dbt.contracts.graph.unparsed import NodeVersion
+from dbt.contracts.graph.nodes import ManifestOrPublicNode
+from dbt.node_types import NodeType, AccessType
+
+
+@dataclass
+class ProjectDependency(dbtClassMixin):
+    name: str
+    environment: str
+
+
+@dataclass
+class ProjectDependencies(dbtClassMixin):
+    projects: List[ProjectDependency] = field(default_factory=list)
+
+
+@dataclass
+class PublicationMetadata(BaseArtifactMetadata):
+    dbt_schema_version: str = field(
+        default_factory=lambda: str(PublicationArtifact.dbt_schema_version)
+    )
+    adapter_type: Optional[str] = None
+    quoting: Dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass
+class PublicModel(dbtClassMixin, ManifestOrPublicNode):
```

Comment:
hmm - what about `ProjectDependencyModel`, or simply `CrossProjectModel`? To avoid ambiguity with internal models that are public.

Reply:
Would we change the name of the "public_nodes" dictionary then?

Reply:
Also the same class is used to construct the public_models in the publication artifact.

Reply:
> Also the same class is used to construct the public_models in the publication artifact.

Ah, okay! I hadn't realized this when starring at this name in isolation. In that case I think PublicModel is fine 👍